imagedephi/redact/tiff.py

```diff
@@ -82,26 +82,46 @@ def _iter_tiff_tag_entries(
     def __init__(
         self,
         tiff_info: TiffInfo,
-        base_rule_set: RuleSet,
-        override_rule_set: RuleSet | None = None,
+        base_rule_set: Ruleset,
+        override_rule_set: Ruleset | None = None,
     ) -> None:
         self.tiff_info = tiff_info
 
         self.redaction_steps = {}
         self.no_match_tags = []
         ifds = self.tiff_info["ifds"]
-        override_rules = override_rule_set.get_metadata_tiff_rules() if override_rule_set else []
-        base_rules = base_rule_set.get_metadata_tiff_rules()
+        override_rules = (
+            override_rule_set.get_format_rules(self.file_format).metadata
+            if override_rule_set
+            else None
+        )
```

Comment:
Since `TiffMetadataRedactionPlan.__init__` can be called by subclasses, I wanted a way to get the right format rules from a `Ruleset`, which is why `Ruleset.get_format_rules` exists. It does feel kind of hacky, so if there's a better way to accomplish this, please let me know. @brianhelba